std/string.d

```diff
@@ -257,6 +257,16 @@ inout(wchar)[] fromStringz(inout(wchar)* cString) @nogc @system pure nothrow
     assert(fromStringz(foo) == null);
     assert(fromStringz("foo"w.ptr) == "foo"w);
     assert(fromStringz("foo\0"w.ptr) == "foo"w);
+    assert(fromStringz("福\0"w.ptr) == "福"w);
+}
+
+@system pure unittest
+{
+    // Test for all surrogate pairs
+    // (its a total of 1048576 itterations but its fast enough)
```

Comment:
This seems excessive for something that's going to be run every time. Either test for specific failure scenarios, or just test endpoints and midpoints.

Reply:
Hmm, we have similar tests in std.algorithm (e.g. remove) or std.datetime and 1m iterations shouldn't take too long. It probably is still faster than a single test in std.concurrency
I would only worry about this if the execution time is really noticable/measurable.

Reply:
@n8sh I agree it is excessive, but since it takes ~0.2ms to run I thought it would be better to be comprehensive. It is also hard for me to imagine corner cases in which wsclen and strlen16 would behave differently so this was to cover as many cases as possible given your suggestion.

Reply:
Why merely all surrogate pairs? Why not all sequences of two consecutive `wchar`s? A single test is sufficient to verify that the custom `wcslen` inside `fromStringz` is not performing auto-decoding. What are the subsequent million checks testing?

Reply:
>What are the subsequent million checks testing?

The speed of electrons?

Reply:
@n8sh You are right I will consider revising the test, but having worked in the industry for 15 years I think the world would be a better place with more comprehensive tests that saving some CPU cycles ;)